library/core/src/fmt/rt.rs

```diff
@@ -150,8 +154,12 @@ impl Argument<'_> {
         Self::new(x, UpperExp::fmt)
     }
     #[inline]
+    #[track_caller]
     pub const fn from_usize(x: &usize) -> Argument<'_> {
-        Argument { ty: ArgumentType::Count(*x) }
+        if *x > u16::MAX as usize {
```

Comment:
Perhaps we should special-case `usize::MAX` and treat it as "unlimited".  This is what was done in `stream-rust`, apparently.  ISTM that that wasn't wrong - indeed, it's convenient.  We shouldn't break it.

Reply:
In `stream-rust`, values close to usize::MAX were used, not usize::MAX: it used `usize::MAX - n`, where `n` was the length of the string. And as of https://github.com/vasekp/stream-rust/commit/f6ee7e8975ae7b15a7e363a7ec803fb06ed995dc, it no longer uses such high values at all.

The crater run hasn't found any cases where anybody uses usize::MAX or values close to it.